UM/Decorators.py

```diff
@@ -89,3 +84,6 @@ def immutable(cls):
     property_names = list(filter(lambda i: isinstance(i, property), inspect.getmembers(cls)))
     cls.__slots__ = property_names
     return cls
+
+def sameSignature(a: inspect.Signature, b: inspect.Signature) -> bool:
+    return len(a.parameters) == len(b.parameters)
```

Comment:
It is good enough until we remove it all and rely on mypy. I had to change this part of the code to allow methods which are the same to match in the case where one has types and other doesn't.